src/topology/category/Top/limits.lean

```diff
@@ -125,6 +126,393 @@ instance forget_preserves_colimits : preserves_colimits (forget : Top.{u} ⥤ Ty
+    (pullback_map_embedding_of_embeddings f g f (𝟙 _)
+      (homeo_of_iso (iso.refl _)).embedding H (𝟙 _) rfl (by simp)),
+  erw ←coe_comp,
+  simp
+end
+
+lemma embedding_of_pullback_embeddings {X Y S : Top}
+  {f : X ⟶ S} {g : Y ⟶ S} (H₁ : embedding f) (H₂ : embedding g) :
+  embedding (limit.π (cospan f g) walking_cospan.one) :=
+begin
+  convert H₂.comp (snd_embedding_of_left_embedding H₁ g),
+  erw ←coe_comp,
+  congr,
+  exact (limit.w _ walking_cospan.hom.inr).symm
+end
+
+lemma snd_open_embedding_of_left_open_embedding {X Y S : Top}
+  {f : X ⟶ S} (H : open_embedding f) (g : Y ⟶ S) :
+  open_embedding ⇑(pullback.snd : pullback f g ⟶ Y) :=
+begin
+  convert (homeo_of_iso (as_iso (pullback.snd : pullback (𝟙 S) g ⟶ _))).open_embedding.comp
+    (pullback_map_open_embedding_of_open_embeddings f g (𝟙 _) g H
+      (homeo_of_iso (iso.refl _)).open_embedding (𝟙 _) rfl (by simp)),
+  erw ←coe_comp,
+  simp
+end
+
+lemma fst_open_embedding_of_right_open_embedding {X Y S : Top}
+  (f : X ⟶ S) {g : Y ⟶ S} (H : open_embedding g) :
+  open_embedding ⇑(pullback.fst : pullback f g ⟶ X) :=
+begin
+  convert (homeo_of_iso (as_iso (pullback.fst : pullback f (𝟙 S) ⟶ _))).open_embedding.comp
+    (pullback_map_open_embedding_of_open_embeddings f g f (𝟙 _)
+      (homeo_of_iso (iso.refl _)).open_embedding H (𝟙 _) rfl (by simp)),
+  erw ←coe_comp,
+  simp
+end
+
+lemma open_embedding_of_pullback_open_embeddings {X Y S : Top}
+  {f : X ⟶ S} {g : Y ⟶ S} (H₁ : open_embedding f) (H₂ : open_embedding g) :
+  open_embedding (limit.π (cospan f g) walking_cospan.one) :=
+begin
+  convert H₂.comp (snd_open_embedding_of_left_open_embedding H₁ g),
+  erw ←coe_comp,
+  congr,
+  exact (limit.w _ walking_cospan.hom.inr).symm
+end
+
+end pullback
+lemma colimit_coinduced (F : J ⥤ Top.{u}) :
+  (colimit F).topological_space = ⨆ j, (F.obj j).topological_space.coinduced (colimit.ι F j) :=
+begin
+  let homeo := homeo_of_iso (colimit.iso_colimit_cocone ⟨_, colimit_cocone_is_colimit F⟩),
+  ext,
+  refine homeo.symm.is_open_preimage.symm.trans (iff.trans _ is_open_supr_iff.symm),
+  exact is_open_supr_iff
+end
+
+lemma colimit_is_open_iff (F : J ⥤ Top.{u}) (U : set ((colimit F : _) : Type u)) :
+  is_open U ↔ ∀ j, is_open (colimit.ι F j ⁻¹' U) :=
+begin
+  conv_lhs { rw colimit_coinduced F },
+  exact is_open_supr_iff
+end
+
+lemma coequalizer_is_open_iff (F : walking_parallel_pair ⥤ Top.{u})
+  (U : set ((colimit F : _) : Type u)) :
+  is_open U ↔ is_open (colimit.ι F walking_parallel_pair.one ⁻¹' U) :=
+begin
+  rw colimit_is_open_iff,
+  split,
+  { intro H, exact H _ },
+  { intros H j,
+    cases j,
+    { rw ←colimit.w F walking_parallel_pair_hom.left,
+      exact (F.map walking_parallel_pair_hom.left).continuous_to_fun.is_open_preimage _ H },
+    { exact H } }
+end
+
+/-- The isomorphism `∐ α ≅ Σ i, α i` as types. -/
+def sigma_iso_sigma {ι : Type u} (α : ι → Top) : ((∐ α : Top) : Type*) ≅ Σ i, α i :=
```

Comment:
Again, I think it would be more useful to have a topological isomorphism here of the form
```lean
def sigma_iso_sigma {ι : Type u} (α : ι → Top.{u}) : ∐ α ≅ Top.of (Σ i, α i) := sorry
```